os/lib/assert.h

```diff
@@ -52,9 +52,7 @@ void _xassert(const char *, int) CC_NORETURN;
 #endif /* NDEBUG */
 
 #ifndef CTASSERT                /* Allow lint to override */
-#define CTASSERT(x)             _CTASSERT(x, __LINE__)
-#define _CTASSERT(x, y)         __CTASSERT(x, y)
-#define __CTASSERT(x, y)        typedef char __assert ## y[(x) ? 1 : -1]
+#define CTASSERT(x) _Static_assert(x, #x)
```

Comment:
This is C11. Both arm and msp430 have build flags to use older standards than that.

Reply:
Should we aim to continue to support them? The build checks, with the default compiler options, are passing.

Reply:
Alternatively, can you suggest our own implementation of CTASSERT than would compile with the current build options and a recent compiler?

Reply:
I haven't tried, but Arm could probably be switched to C11.

Somewhat surprisingly, it seems msp430-gcc accepts this:
```
$ cat test.c 
_Static_assert(3 < 4, "yope");
_Static_assert(3 > 4, "nope");
$ msp430-gcc -Wall test.c -c -std=gnu11
test.c:2:1: error: static assertion failed: "nope"
```
It doesn't have an `assert.h` include though, so the C11 compliance is limited.

Reply:
It seems msp430-gcc supports `_Static_assert` even with `-std=gnu99`, which I guess is fair since it might be a GNU extension.

What is more surprising to me is that arm-none-eabi-gcc supports it with `-std=c99`. Do you care about non-gcc-ARM-compilers? They might be more strict with enforcing c99.

Reply:
> Do you care about non-gcc-ARM-compilers? 
IMO we already use GCC specific extensions. For example, the statement-as-expression idiom in the `RTIMER_BUSYWAIT_UNTIL_ABS` implementation. So, probably no.

Reply:
Could it be an idea to move the CTASSERT bit in a separate PR so that it is not blocking the ETX changes? You could use the preprocessor for now?

Reply:
@arurke done!